simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/GetBaseReportRequestHandler.java

```diff
@@ -0,0 +1,64 @@
+package com.evbox.everon.ocpp.simulator.station.handlers.ocpp;
+
+import com.evbox.everon.ocpp.simulator.station.StationMessageSender;
+import com.evbox.everon.ocpp.simulator.station.component.StationComponentsHolder;
+import com.evbox.everon.ocpp.v20.message.station.GetBaseReportRequest;
+import com.evbox.everon.ocpp.v20.message.station.GetBaseReportResponse;
+import com.evbox.everon.ocpp.v20.message.station.ReportDatum;
+
+import javax.annotation.Nullable;
+import java.time.Clock;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.evbox.everon.ocpp.v20.message.station.GetBaseReportResponse.Status.ACCEPTED;
+import static com.evbox.everon.ocpp.v20.message.station.GetBaseReportResponse.Status.NOT_SUPPORTED;
+import static java.time.ZonedDateTime.ofInstant;
+import static java.util.Collections.singletonList;
+
+public class GetBaseReportRequestHandler implements OcppRequestHandler<GetBaseReportRequest> {
+
+    private final StationMessageSender stationMessageSender;
+    private final StationComponentsHolder stationComponentsHolder;
+    private final Clock clock;
+
+    public GetBaseReportRequestHandler(Clock clock, StationComponentsHolder stationComponentsHolder, StationMessageSender stationMessageSender) {
+        this.stationMessageSender = stationMessageSender;
+        this.stationComponentsHolder = stationComponentsHolder;
+        this.clock = clock;
+    }
+
+    @Override
+    public void handle(String callId, GetBaseReportRequest request) {
+        switch (request.getReportBase()) {
```

Comment:
Please refactor this code to avoid conditional-logic

Reply:
how?

Reply:
it is async by it's nature. csms sends getbasereport and station replies asynchronously with notification request. no need to add any additional logic

Reply:
You can use polymorphism to do that. For example:

```
class abstract ReportBase {
     abstract void sendReport(callId, request);
}

class FullInventoryReportBase extends ReportBase {
   // impl...
}
```

now in your handler declare a map, such that:

`Map<Status, ReportBase> = of(entry(Full_INVENTORY, new FullInventoryBaseReport()), ...);
`
and your handle method would look:
`reportBaseMap.get(request.getReportBase()).sendReport(callId, request);
`

Reply:
done